src/qt/bitcoinunits.cpp

```diff
@@ -182,9 +197,13 @@ QString BitcoinUnits::floorWithUnit(int unit, const CAmount& amount, bool plussi
     return result + QString(" ") + name(unit);
 }
 
-QString BitcoinUnits::floorHtmlWithUnit(int unit, const CAmount& amount, bool plussign, SeparatorStyle separators)
+QString BitcoinUnits::floorHtmlWithPrivacy(int unit, const CAmount& amount, SeparatorStyle separators, bool privacy)
 {
-    QString str(floorWithUnit(unit, amount, plussign, separators));
+    assert(amount >= 0);
+    QString str = privacy
+        ? floorWithUnit(unit, 0, false, separators).replace('0', '#')
+        : floorWithUnit(unit, amount, false, separators);
+
```

Comment:
16432: note, this differs from the original PR

Reply:
bitcoin in GUI show all 8 digits after dot everytime but dash does floor to 2 sign after decimal dot.
Due to that this backport is not 100% same but adopted for `floorHtmlWithPrivacy` instead `formatWithPrivacy` indeed.

But this different is expected to be.

Reply:
makes sense